Bug: Failed to build windows-sys v0.28 for UWP targets

### Which crate is this about?

windows-sys

### Crate version

0.28.0

### Summary

```toml
[package]
name = "testlib"
version = "0.1.0"
authors = ["bdbai <bdbaiapp@163.com>"]
edition = "2021"

[dependencies]
windows-sys = { version = "0.28", features = ["Foundation_Collections"] }
```
```rust
fn main() {
    #[allow(unused_imports)]
    use windows_sys;
}
```
```powershell
cargo build -Z build-std=std,panic_abort --target x86_64-uwp-windows-msvc
```

### Expected behavior

The program compiles successfully.

### Actual behavior

   Compiling testlib v0.1.0 (D:\proj\testrs\testlib)
error: linking with `link.exe` failed: exit code: 1181
  |
  = note: "D:\\soft\\vs2022\\VC\\Tools\\MSVC\\14.30.30705\\bin\\HostX64\\x64\\link.exe" "/LIBPATH:D:\\soft\\vs2022\\VC\\Tools\\MSVC\\14.30.30705\\lib\\x64\\store" "/NOLOGO" "/APPCONTAINER" "mincore.lib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.12vmw4gj9nfwec88.rcgu.o" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.1j1n6no1c99akr0j.rcgu.o" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.1jk4kivmwworo4w.rcgu.o" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.27gpty3m03fap44q.rcgu.o" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.4232um37ftko5bak.rcgu.o" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.52aknw0u11xqg9o0.rcgu.o" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.3y3ebs8hltnhz6at.rcgu.o" "/LIBPATH:D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps" "/LIBPATH:D:\\proj\\testrs\\testlib\\target\\debug\\deps" "/LIBPATH:C:\\Users\\bdbai\\.rustup\\toolchains\\nightly-2022-01-01-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-uwp-windows-msvc\\lib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libwindows_sys-83302516282eaf16.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libstd-887136ed81f7457d.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libpanic_unwind-0c7692820fe96ff7.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libstd_detect-121b5eaf2b92134f.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\librustc_demangle-64710a2d26a0f79a.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libhashbrown-d723a7a0db3770a4.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\librustc_std_workspace_alloc-c1810ac5ca88d78b.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libunwind-012d9bb6d5e8961d.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libcfg_if-9d3295dd49b34d62.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\liblibc-3e453c293f500f34.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\liballoc-7100162ff8163776.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\librustc_std_workspace_core-ca4d8718395fe3fa.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libcore-f5e35ad5f4b3cb6d.rlib" "D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\libcompiler_builtins-358995a960b47c60.rlib" "windows.lib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\Users\\bdbai\\.rustup\\toolchains\\nightly-2022-01-01-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-uwp-windows-msvc\\lib" "/OUT:D:\\proj\\testrs\\testlib\\target\\x86_64-uwp-windows-msvc\\debug\\deps\\testlib.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\bdbai\\.rustup\\toolchains\\nightly-2022-01-01-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\bdbai\\.rustup\\toolchains\\nightly-2022-01-01-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\bdbai\\.rustup\\toolchains\\nightly-2022-01-01-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\bdbai\\.rustup\\toolchains\\nightly-2022-01-01-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: LINK : fatal error LNK1181: cannot open input file 'windows.lib'


error: could not compile `testlib` due to previous error

### Additional comments

Seems like `windows_x86_64_msvc` was not picked for `x86_64-uwp-windows-msvc` target, and the build script for the targets did not recognize `x86_64-uwp-windows-msvc` as well.

https://github.com/microsoft/windows-rs/blob/dfc25285a289847ad88cbd206d169f7697be0dd4/crates/targets/x86_64_msvc/build.rs#L2

--- a/crates/libs/sys/Cargo.toml
+++ b/crates/libs/sys/Cargo.toml
@@ -16,16 +16,31 @@ all-features = true
 [target.i686-pc-windows-msvc.dependencies]
 windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
 
+[target.i686-uwp-windows-msvc.dependencies]
+windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
+
 [target.x86_64-pc-windows-msvc.dependencies]
+windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-msvc.dependencies]
 windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
 
 [target.aarch64-pc-windows-msvc.dependencies]
 windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
 
+[target.aarch64-uwp-windows-msvc.dependencies]
+windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
+
 [target.i686-pc-windows-gnu.dependencies]
 windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
 
+[target.i686-uwp-windows-gnu.dependencies]
+windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
+
 [target.x86_64-pc-windows-gnu.dependencies]
+windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-gnu.dependencies]
 windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
 
 [features]

--- a/crates/libs/windows/Cargo.toml
+++ b/crates/libs/windows/Cargo.toml
@@ -17,16 +17,31 @@ targets = []
 [target.i686-pc-windows-msvc.dependencies]
 windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
 
+[target.i686-uwp-windows-msvc.dependencies]
+windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
+
 [target.x86_64-pc-windows-msvc.dependencies]
+windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-msvc.dependencies]
 windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
 
 [target.aarch64-pc-windows-msvc.dependencies]
 windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
 
+[target.aarch64-uwp-windows-msvc.dependencies]
+windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
+
 [target.i686-pc-windows-gnu.dependencies]
 windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
 
+[target.i686-uwp-windows-gnu.dependencies]
+windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
+
 [target.x86_64-pc-windows-gnu.dependencies]
+windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-gnu.dependencies]
 windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
 
 [dependencies]

--- a/crates/targets/aarch64_msvc/build.rs
+++ b/crates/targets/aarch64_msvc/build.rs
@@ -1,5 +1,6 @@
 fn main() {
-    if std::env::var("TARGET").unwrap() != "aarch64-pc-windows-msvc" {
+    let target = std::env::var("TARGET").unwrap();
+    if target != "aarch64-pc-windows-msvc" && target != "aarch64-uwp-windows-msvc" {
         return;
     }
 

--- a/crates/targets/i686_gnu/build.rs
+++ b/crates/targets/i686_gnu/build.rs
@@ -1,5 +1,6 @@
 fn main() {
-    if std::env::var("TARGET").unwrap() != "i686-pc-windows-gnu" {
+    let target = std::env::var("TARGET").unwrap();
+    if target != "i686-pc-windows-gnu" && target != "i686-uwp-windows-gnu" {
         return;
     }
 

--- a/crates/targets/i686_msvc/build.rs
+++ b/crates/targets/i686_msvc/build.rs
@@ -1,5 +1,6 @@
 fn main() {
-    if std::env::var("TARGET").unwrap() != "i686-pc-windows-msvc" {
+    let target = std::env::var("TARGET").unwrap();
+    if target != "i686-pc-windows-msvc" && target != "i686-uwp-windows-msvc" {
         return;
     }
 

--- a/crates/targets/x86_64_gnu/build.rs
+++ b/crates/targets/x86_64_gnu/build.rs
@@ -1,5 +1,6 @@
 fn main() {
-    if std::env::var("TARGET").unwrap() != "x86_64-pc-windows-gnu" {
+    let target = std::env::var("TARGET").unwrap();
+    if target != "x86_64-pc-windows-gnu" && target != "x86_64-uwp-windows-gnu" {
         return;
     }
 

--- a/crates/targets/x86_64_msvc/build.rs
+++ b/crates/targets/x86_64_msvc/build.rs
@@ -1,5 +1,6 @@
 fn main() {
-    if std::env::var("TARGET").unwrap() != "x86_64-pc-windows-msvc" {
+    let target = std::env::var("TARGET").unwrap();
+    if target != "x86_64-pc-windows-msvc" && target != "x86_64-uwp-windows-msvc" {
         return;
     }
 

--- a/crates/tools/api/src/main.rs
+++ b/crates/tools/api/src/main.rs
@@ -41,16 +41,31 @@ targets = []
 [target.i686-pc-windows-msvc.dependencies]
 windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
 
+[target.i686-uwp-windows-msvc.dependencies]
+windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
+
 [target.x86_64-pc-windows-msvc.dependencies]
+windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-msvc.dependencies]
 windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
 
 [target.aarch64-pc-windows-msvc.dependencies]
 windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
 
+[target.aarch64-uwp-windows-msvc.dependencies]
+windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
+
 [target.i686-pc-windows-gnu.dependencies]
 windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
 
+[target.i686-uwp-windows-gnu.dependencies]
+windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
+
 [target.x86_64-pc-windows-gnu.dependencies]
+windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-gnu.dependencies]
 windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
 
 [dependencies]

--- a/crates/tools/sys/src/main.rs
+++ b/crates/tools/sys/src/main.rs
@@ -40,16 +40,31 @@ all-features = true
 [target.i686-pc-windows-msvc.dependencies]
 windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
 
+[target.i686-uwp-windows-msvc.dependencies]
+windows_i686_msvc = { path = "../../targets/i686_msvc", version = "0.29.0" }
+
 [target.x86_64-pc-windows-msvc.dependencies]
+windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-msvc.dependencies]
 windows_x86_64_msvc = { path = "../../targets/x86_64_msvc", version = "0.29.0" }
 
 [target.aarch64-pc-windows-msvc.dependencies]
 windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
 
+[target.aarch64-uwp-windows-msvc.dependencies]
+windows_aarch64_msvc = { path = "../../targets/aarch64_msvc", version = "0.29.0" }
+
 [target.i686-pc-windows-gnu.dependencies]
 windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
 
+[target.i686-uwp-windows-gnu.dependencies]
+windows_i686_gnu = { path = "../../targets/i686_gnu", version = "0.29.0" }
+
 [target.x86_64-pc-windows-gnu.dependencies]
+windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
+
+[target.x86_64-uwp-windows-gnu.dependencies]
 windows_x86_64_gnu = { path = "../../targets/x86_64_gnu", version = "0.29.0" }
 
 [features]

